src/qibolab/result.py

```diff
@@ -114,20 +114,15 @@ class StateResults:
     Associated with AcquisitionType.DISCRIMINATION and AveragingMode.SINGLESHOT
     """
 
-    def __init__(self, states: np.ndarray = np.array([]), nshots=None):
-        self.states: Optional[npt.NDArray[np.uint32]] = (
-            states.reshape(states.shape[0] // nshots, nshots) if nshots else states
-        )
-
-    @property
-    def states(self):
-        return self._states
+    states: npt.NDArray[np.uint32]
+    nshots: Optional[int] = None
 
-    @states.setter
-    def states(self, values):
-        if not np.all((values >= 0) & (values <= 1)):
+    def __post_init__(self, states: np.ndarray = np.array([]), nshots=None):
+        if not np.all((self.states >= 0) & (self.states <= 1)):
```

Comment:
Do we really need this check? If I understood correctly `states` should be an array of 0s and 1s, right?